pkg/agent/datamodel/types.go

```diff
@@ -1606,6 +1606,10 @@ type K8sComponents struct {
 	// Full path to the Windows package (windowszip) to use.
 	// For example: https://acs-mirror.azureedge.net/kubernetes/v1.17.8/windowszip/v1.17.8-1int.zip.
 	WindowsPackageURL string
+
+	// Full path to the Linux package (tar.gz) to use.
+	// For example: url=https://acs-mirror.azureedge.net/kubernetes/v1.25.6-hotfix.20230612/binaries/v1.25.6-hotfix.20230612.tar.gz
+	LinuxPrivatePackageURL string
```

Comment:
I guess I'm a bit confused why we really need this and can't just use something like a boolean flag `usePrivateKubePackages` since the URL effectively isn't actually used other than to tell CSE to first check that there are private packages cached on the VHD and use them if they're there

is it mainly because it would require changes on the RP side?

Reply:
It is because private package name will come from RP side, the kubernetes_version could be 1.28.3 for example, but the package could be v1.28.3-hotfix.20231103.tgz or v1.28.3.tgz etc.

Reply:
can you add a note here to mention that this URL is _only_ used at build-time outside of CSE, and this is only used as a marker to check whether private pkgs are cached at runtime?